### PR TITLE
Return else_block_input_var_id from lower_optional_else_block.

### PR DESCRIPTION
**Pull request by ilyalesokhin-starkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2417)

### CODE REVIEW

**Review by ilyalesokhin-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2417)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @orizi and @spapinistarkware)

___
*[`crates/cairo-lang-lowering/src/lower/lower_if.rs` line 87 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2417#-NQ0Hu2D6DAxKnXWKGno:-NQ0Hu2D6DAxKnXWKGnp:b-wigd8j) ([raw file](https://github.com/starkware-libs/cairo/blob/38d7709931bff8f54e2cbe4bf5cd4d0f01ad60d9/crates/cairo-lang-lowering/src/lower/lower_if.rs#L87)):*
> ```XML
>     let subscope_else = create_subscope_with_bound_refs(ctx, scope);
>     let block_else_id = subscope_else.block_id;
>     let (_else_block_input_var_id, block_else) =
> ```

I'll need this to generate the MatchArm in the next PR, 
I'm not sure what is the cleanest way of extracting it.



_Code quote:_
```XML
_else_block_input_var_id
```

**Review by ilyalesokhin-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2417)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @orizi and @spapinistarkware)

___
*[`crates/cairo-lang-lowering/src/lower/lower_if.rs` line 87 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2417#-NQ0Hu2D6DAxKnXWKGno:-NQ0IAzP8UQTIxvRdbfA:b-t59gut) ([raw file](https://github.com/starkware-libs/cairo/blob/38d7709931bff8f54e2cbe4bf5cd4d0f01ad60d9/crates/cairo-lang-lowering/src/lower/lower_if.rs#L87)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

I'll need this to generate the MatchArm in the next PR, 
I'm not sure what is the cleanest way of extracting it.
</blockquote></details>

we could always use the `if_location` for this variable.



_Code quote:_
```XML
if_location
```

**Review by ilyalesokhin-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2417)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @orizi and @spapinistarkware)

___
*[`crates/cairo-lang-lowering/src/lower/lower_if.rs` line 87 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2417#-NQ0Hu2D6DAxKnXWKGno:-NQ0JYbrCpFBgJDPuEE2:b-x0adfq) ([raw file](https://github.com/starkware-libs/cairo/blob/38d7709931bff8f54e2cbe4bf5cd4d0f01ad60d9/crates/cairo-lang-lowering/src/lower/lower_if.rs#L87)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

we could always use the `if_location` for this variable.
</blockquote></details>

In other words.

move 
            let else_block_var_id =
                scope.add_input(ctx, VarRequest { ty: else_block_input_ty, location: if_location });

out of lower_optional_else_block.
and use it regardless of if we have an else block or not.



_Code quote:_
```XML
            let else_block_var_id =
                scope.add_input(ctx, VarRequest { ty: else_block_input_ty, location: if_location });
```

**Review by orizi — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2417)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @spapinistarkware)

___
*[`crates/cairo-lang-lowering/src/lower/lower_if.rs` line 87 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2417#-NQ0Hu2D6DAxKnXWKGno:-NQ0KAG87Zd8Y9Rt9kP5:bg8okg0) ([raw file](https://github.com/starkware-libs/cairo/blob/38d7709931bff8f54e2cbe4bf5cd4d0f01ad60d9/crates/cairo-lang-lowering/src/lower/lower_if.rs#L87)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

In other words.

move 
            let else_block_var_id =
                scope.add_input(ctx, VarRequest { ty: else_block_input_ty, location: if_location });

out of lower_optional_else_block.
and use it regardless of if we have an else block or not.


</blockquote></details>

this sounds annoying - but probably required for the move itself

**Review by ilyalesokhin-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2417)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @spapinistarkware)

___
*[`crates/cairo-lang-lowering/src/lower/lower_if.rs` line 87 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2417#-NQ0Hu2D6DAxKnXWKGno:-NQ0Kjjw7hLRYMHn5cXe:b-z9se5a) ([raw file](https://github.com/starkware-libs/cairo/blob/38d7709931bff8f54e2cbe4bf5cd4d0f01ad60d9/crates/cairo-lang-lowering/src/lower/lower_if.rs#L87)):*
<details><summary><i>Previously, orizi wrote…</i></summary><blockquote>

this sounds annoying - but probably required for the move itself
</blockquote></details>

I don't understand, what solution do you prefer?

**Review by orizi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2417)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @spapinistarkware)

___
*[`crates/cairo-lang-lowering/src/lower/lower_if.rs` line 87 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2417#-NQ0Hu2D6DAxKnXWKGno:-NQ0LMgA9xSgjbc1Pp44:box7ee1) ([raw file](https://github.com/starkware-libs/cairo/blob/38d7709931bff8f54e2cbe4bf5cd4d0f01ad60d9/crates/cairo-lang-lowering/src/lower/lower_if.rs#L87)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

I don't understand, what solution do you prefer?
</blockquote></details>

you suggested only a single solution (IIUC) and it sounds good enough.

**Review by ilyalesokhin-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2417)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @spapinistarkware)

___
*[`crates/cairo-lang-lowering/src/lower/lower_if.rs` line 87 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2417#-NQ0Hu2D6DAxKnXWKGno:-NQ0NoiI3mMK35K8Qyjr:b-rekftl) ([raw file](https://github.com/starkware-libs/cairo/blob/38d7709931bff8f54e2cbe4bf5cd4d0f01ad60d9/crates/cairo-lang-lowering/src/lower/lower_if.rs#L87)):*
<details><summary><i>Previously, orizi wrote…</i></summary><blockquote>

you suggested only a single solution (IIUC) and it sounds good enough.
</blockquote></details>

the second option is here:
https://reviewable.io/reviews/starkware-libs/cairo/2418

**Review by orizi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2417)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @spapinistarkware)

___
*[`crates/cairo-lang-lowering/src/lower/lower_if.rs` line 87 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2417#-NQ0Hu2D6DAxKnXWKGno:-NQ0OxVs68TTuJGwGbtC:b-9f6cx5) ([raw file](https://github.com/starkware-libs/cairo/blob/38d7709931bff8f54e2cbe4bf5cd4d0f01ad60d9/crates/cairo-lang-lowering/src/lower/lower_if.rs#L87)):*
<details><summary><i>Previously, ilyalesokhin-starkware wrote…</i></summary><blockquote>

the second option is here:
https://reviewable.io/reviews/starkware-libs/cairo/2418
</blockquote></details>

the second option seems cleaner to me - so i vote for it.

**Comment by ilyalesokhin-starkware** [comment]
did
https://reviewable.io/reviews/starkware-libs/cairo/2418
instead
